src/core/qgstaskmanager.cpp

```diff
@@ -142,6 +142,27 @@ QList<QgsMapLayer *> QgsTask::dependentLayers() const
   return _qgis_listQPointerToRaw( mDependentLayers );
 }
 
+bool QgsTask::waitForFinished( int timeout )
```

Comment:
To not exceed the maximum number of concurrent tasks (I was already considering `waitForFinished( QList<QgsTask* > )` as well ).